### PR TITLE
Parameter to skip validation during Training

### PR DESCRIPTION
**Pull request by corbob**
Validation can take a significant amount of time. This may not be desired for all training.

This introduces the ability to specify if you'd like to validate your model or not.

### CODE REVIEW

File: serpent/machine_learning/context_classification/context_classifier.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def available_implementations(cls):
         ]
 
     @classmethod
-    def executable_train(cls, epochs=3, classifier="CNNInceptionV3ContextClassifier"):
+    def executable_train(cls, epochs=3, classifier="CNNInceptionV3ContextClassifier", validation=True):
```
Comment (nbrochu) [inline serpent/machine_learning/context_classification/context_classifier.py]: You _should_ call this validate plz

File: serpent/machine_learning/context_classification/context_classifier.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def executable_train(cls, epochs=3, classifier="CNNInceptionV3ContextClassifier"
 
             context_classifier = context_classifier_class(input_shape=frame.shape)
 
-            context_classifier.train(epochs=epochs)
+            context_classifier.train(epochs=epochs, validation=validation)
```
Comment (nbrochu) [inline serpent/machine_learning/context_classification/context_classifier.py]: validate

File: serpent/machine_learning/context_classification/context_classifiers/cnn_inception_v3_context_classifier.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def __init__(self, input_shape=None):
         self.training_generator = None
         self.validation_generator = None
 
-    def train(self, epochs=3):
+    def train(self, epochs=3, validation=True):
```
Comment (nbrochu) [inline serpent/machine_learning/context_classification/context_classifiers/cnn_inception_v3_context_classifier.py]: validate

File: serpent/serpent.py [hunk 4]
```diff
@@ -420,9 +420,14 @@ def prepare_game_agent_plugin(game_agent_name):
         f.write(contents)
 
 
-def train_context(epochs=3):
+def train_context(epochs=3, validation="True"):
```
Comment (nbrochu) [inline serpent/serpent.py]: validate

Comment (nbrochu) [reply serpent/serpent.py]: True as bool

File: serpent/serpent.py [hunk 5]
```diff
@@ -420,9 +420,14 @@ def prepare_game_agent_plugin(game_agent_name):
...
     from serpent.machine_learning.context_classification.context_classifier import ContextClassifier
-    ContextClassifier.executable_train(epochs=int(epochs))
+    ContextClassifier.executable_train(epochs=int(epochs), validation=validation.lower() in (
```
Comment (nbrochu) [inline serpent/serpent.py]: validate

Comment (nbrochu) [reply serpent/serpent.py]: validation == "True"

**Review by nbrochu — CHANGES_REQUESTED** [review]

**Comment by corbob** [comment]
Requested changes updated.

**Pull request closed without merging**